addons/html_builder/static/src/builder/plugins/dependency_manager.js

```diff
@@ -0,0 +1,43 @@
+import { EventBus } from "@odoo/owl";
+
+export class DependencyManager extends EventBus {
+    constructor() {
+        super();
+        this.dependencies = [];
+        this.dependenciesMap = {};
+        this.count = 0;
+        this.currentCount = 0;
+    }
+    update() {
+        this.dependenciesMap = {};
+        for (const [id, value] of this.dependencies) {
+            this.dependenciesMap[id] = value;
+        }
+        this.currentCount = this.count;
+    }
+
+    add(id, value) {
+        // In case the dependency is added after a dependent try to get it
+        // an event is scheduled to notify the dependent about it.
+        if (!this.nextAddUpdate) {
+            this.nextAddUpdate = Promise.resolve().then(() => {
+                this.trigger("dependency-added");
+                this.nextAddUpdate = undefined;
+            });
+        }
```

Comment:
Could you explain ?

Reply:
There is a comment just above 
```
        // In case the dependency is added after a dependent try to get it
        // an event is scheduled to notify the dependent about it.
```